### ISSUE TITLE
How to setup babylon react native into expo react native

### ISSUE THREAD

**Issue by fritzzu** [BabylonJS/BabylonReactNative#117]
Hi, I can't setup babylon react native into expo react native project, I got this error:

Module parse failed: The keyword 'enum' is reserved (12:0)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
| // https://developer.mozilla.org/en-US/docs/Web/API/XRSystem/requestSession
| // https://developer.mozilla.org/en-US/docs/Web/API/DOMException#Error_names
> enum DOMError {
|     NotSupportedError = 9,
|     InvalidStateError = 11,

I added to package.json: 
    "@babylonjs/core": "^4.2.0",
    "@babylonjs/react-native": "^0.3.0",
    "@babylonjs/loaders": "^4.2.0",
and then code taken from App.tsx from playground.
Thanks.

**Comment by bghgary** [comment]
Is there any way you can set up a repro? It's difficult to help without it.

**Comment by fritzzu** [comment]
You can clone https://github.com/fritzzu/babylon-expo-test , then launch npm i and npm start. You need to launch npm install --global expo-cli first if you don't have it.
When you run npm start, then choose "run in web browser" option (or press letter w in the console) and you'll go the error.
Thanks.

**Comment by seand88** [comment]
i would imagine this wouldnt work with expo? I believe you only get access to the graphics subsystem with expo-gl which this doesnt use. Maybe though detached it would work?

**Comment by fritzzu** [comment]
I still can't use babylon react native with Expo. Maybe with plain React Native it should run flawless. I'm waiting for some hint with Expo.

**Comment by bghgary** [comment]
We might not be able to help that much. Not many people working this project have experience with expo (or expo react native).

Can you explain why you want to use Babylon React Native with expo? It seems like there will be issues even if we can set it up.

**Comment by fritzzu** [comment]
Because I have a Expo application with a webview and I'd like to switch from webview to Babylon React Native without eject from Expo.

**Comment by ryantrem** [comment]
Hi @fritzzu - as @seand88 mentioned above, it likely would work in "ejected" mode, but not in regular Expo mode. From the [Expo docs](https://docs.expo.io/expokit/eject/) regarding "ejecting to ExpoKit":

> Normally, Expo apps are written in pure JS and never "drop down" to the native iOS or Android layer. This is core to the Expo philosophy and it's part of what makes Expo fast and powerful to use.
However, there are some cases where advanced developers need native capabilities outside of what Expo offers out-of-the-box. The most common situation is **when a project requires a specific Native Module which is not supported by React Native Core or the Expo SDK**.

I've never used Expo myself, only read about it. My understanding is that in normal Expo mode, all the native compilation is server side, and so only native modules that (as the docs say) are part of React Native Core and the Expo SDK can be used. "Ejecting" is essentially converting your Expo app into a regular React Native app that you compile locally, and additionally has access to the Expo SDK.

In summary, the thing you want to do (use Babylon React Native in a non-ejected Expo app) is not supported by Expo itself (it is not a limitation of Babylon React Native, and therefore not something we can "fix"). I think if you want to use Babylon React Native in your project, you will have to "eject" your app and compile locally.

**Comment by bghgary** [comment]
Closing due to lack of activity. Please reopen or use the forums to discuss if needed.
